compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2505,7 +2505,29 @@ impl<'a> Parser<'a> {
             // ...now that we've parsed the first argument, `self` is no longer allowed.
             first_param = false;
             param
-        })?;
+        });
+        let mut params = match parsed {
```

Comment:
I was originally doing it after to avoid trying to recover before we'd even encountered an error, and to avoid perf impact, but hopefully with inlining it should be okay